src/code/z_play.c

```diff
@@ -1620,34 +1623,40 @@ s32 func_800C0808(PlayState* this, s16 camId, Player* player, s16 setting) {
     return Camera_ChangeSetting(camera, setting);
 }
 
-s32 Play_CameraChangeSetting(PlayState* this, s16 camId, s16 setting) {
+s32 Play_ChangeCameraSetting(PlayState* this, s16 camId, s16 setting) {
     return Camera_ChangeSetting(Play_GetCamera(this, camId), setting);
 }
 
-void func_800C08AC(PlayState* this, s16 camId, s16 arg2) {
+/**
+ * Smoothly return control from a sub camera to the main camera by moving the subCamera's eye, at, fov through
+ * interpolation from the initial subCam viewParams to the target mainCam viewParams over `duration`.
+ * Setting the `duration` to 0 or less will instantly return control to the main camera.
```

Comment:
I understand what you mean, but afaict control is returned instantly to the main camera, and the animation relies on the 1020 onepointdemo playing on the main camera

Maybe
```
 * Setting the `duration` to 0 or less will instantly snap to the main camera without animation.
```
idk

and "moving the subCamera's eye, at, fov", is just wrong? maybe just don't need to mention how the implementation works

```
 * Smoothly return the view from a sub camera to the main camera by moving the eye, at, fov through
 * interpolation from the current to the target mainCam's over `duration`.
```

:thinking: ?

Reply:
In `OnePointCutscene` 1020 (i.e. when duration is `> 0`), a new subCam is allocated after all subCamera's are cleared. This new subcamera is used to smoothly move from the current view (which was set at the end of the last subCamera) to the main camera:

```c
        case 1020:
            if (timer < 20) {
                timer = 20;
            }
            D_801208EC[0].atTargetInit = play->view.at;
            D_801208EC[0].eyeTargetInit = play->view.eye;
            D_801208EC[0].fovTargetInit = play->view.fovy;
            D_801208EC[1].atTargetInit = mainCam->at;
            D_801208EC[1].eyeTargetInit = mainCam->eye;
            D_801208EC[1].fovTargetInit = mainCam->fov;
            D_801208EC[1].timerInit = timer - 1;
            subCam->timer = timer + 1;
            D_801208EC[1].lerpStepScale = 1.0f / (0.5f * timer);

            csInfo->keyFrames = D_801208EC;
            csInfo->keyFrameCnt = 3;

            Play_InitCameraSettings(play, subCamId, player, CAM_SET_CS_C);
            break;
```
So when duration is > 0, control isn't handed over to the mainCam (it isn't set to active status), instead a new subCam is created through onePointDemo:

That's my understanding at least

Reply:
I completely missed it created a subcam, my bad

I guess you can disregard the whole comment